src/validate-params.ts

```diff
@@ -2,22 +2,22 @@ import {joinErrors, toResult, Result} from './util';
 import * as chalk from 'chalk';
 import SoyContext from './soy-context';
 import JSContext from './js-context';
+import {Config} from './config';
 
-export default function validateParams(soyContext: SoyContext, jsContext: JSContext): Result {
-  /**
-   * We just skip validation if the default class does not extend from Component.
-   * TODO: See if we can resolve and parse the parent class' STATE.
-   */
-  if (jsContext.getSuperClassImportPath() !== 'metal-component') {
-    return toResult(true);
-  }
+function getImplicitParams(className: string | null, paramsMap: any):string[] {
+  return className ? Object.keys(paramsMap)
+    .filter(regex => className.match(new RegExp(regex)))
```

Comment:
We're not safety catching `RegExp` creation exceptions here. We don't do it for the `callToImport` regexes, so not sure what you'd prefer here.

It is true that an invalid regex will cut the iteration producing a partially resolved result, which could cause a lot of headaches when debugging if we fail to announce it...